src/Command/Command.php

```diff
@@ -227,13 +237,22 @@ protected function _formatSummary(array $summary, int $depth = 0) : string {
   /**
    * Gets an API Endpoint instance.
    *
+   * Note, this method hints it returns "Endpoint",
```

Comment:
If we move away from common names like EndPoint for each section, this would probably solve phan's problem.

Reply:
phan understands the classname. This error is because the return typehint is for the _parent_ Endpoint class, which doesn't have the method(s) we may be using on the child class. 

phan is technically correct, here — we're assuming that we get an instance of (e.g.) CloudAccount\Endpoint, when all the return type guarantees is that it's a Resource\Endpoint.  We "know" we will have the proper subclass, but there's nothing in the code that _proves_ that for static analysis (that's what the assert does - it proves we have the right subclass).

Reply:
note also this is from rebasing on master so we're up-to-date — it's not new code